### ISSUE TITLE
CarPlay stopped (overnight)

### ISSUE THREAD

**Issue by scott030871** [thekirankumar/carstream-android-auto#162]
Google appear to be hitting non-google Android Auto Apps.
First they role out Android Auto V3 which makes us roll back to V2.9 to get Carplay working again.
Overnight my Carplay, AA Mirror and OBD2AA stop working on V2.9
How they manage it with Auto Updates turned off amazes me

**Comment by michelesilla** [comment]
Hi guys,

I've the same issue....  AA 2.9 works fine until overnight , then it stop to works togheter widgets like carstream and aamirror ......   Autoupdate was turned off in setting of android play store . How is it possibile??.

Can we fix it?

Best Regards


I've tried to unistall AA 2.9  + carstream + AAmirror and install them by ADB TOOL  but no works.
I,ve tried yo install AA 3.0.570553 same issue..
I'm sure that this is caused from autoupdate google play service from 12.2.21 release to 12.5.21 release..

Damn... there is no way to fix it? I need of AA ! HELP US :)

**Comment by Zetrox2k** [comment]
yup just had the same issue.  I have the android auto desktop head unit emulator and it works fine but not in car.

wish someone would just make a whole new android auto app. I'd pay for that

**Comment by hsuq510** [comment]
I can confirm AA 2.9 was visibile, but after play service 12.5.21 upgrade stop to appear, I downgraded to 12.2.21 and it works again.

**Comment by hsuq510** [comment]
not easy task, I had to downgrade to stock rom.

**Comment by Zetrox2k** [comment]
FIX FOR LATEST BLOCK

Thanks to hsuq510 for pointing out the reason app was blocked.  Heres how to sort it all out though WITHOUT re flashing rom or rooting your phone.

1. Go to Settings -> Apps

2. Select "Google Play Services"

3. Hit the 3 Dots on the top right and select "Uninstall Updates"

4. You should get an error about device administrators

5. click "Manage Device Administrators"

6. Untick any administrators checked (the ones that need to be disabled are find my phone, and if its there, Android Device Manager)

This should now let you uninstall the updates and revert back to your stock version of Google Play Services

7. Now you can download the apk update Version 12.2.21 from apk mirror
https://www.apkmirror.com/apk/google-inc/google-play-services/google-play-services-12-2-21-release/

Tested on S8, and S7

**Comment by air-hun** [comment]
And how to force to don't update again automatically?

**Comment by Zetrox2k** [comment]
air-hun just turn off automatic updates

**Comment by Zetrox2k** [comment]
just a quick note guys once you remove administrator devices it takes around 30 seconds to take effect. so if it still has the same error, wait 30 secs then uninstall updates

**Comment by vegerfrancois** [comment]
I have found it.Works !!

**Comment by air-hun** [comment]
vegerfrancois:
Will you share us the solution?

**Comment by vegerfrancois** [comment]
You must find (find my device) in settings,then you go too apps,google play services and thick the 3 dots in the richt corner,than remove updates.If the updates are removed,you can install the one from apk mirror(as Zetrox2k says)

**Comment by i040628** [comment]
Seems like the app disappeared from the latest button in Android Auto but the app name Carstream Player is showing up for the headset button. Altough I renamed the app. Isn't is the root cause?

**Comment by Zetrox2k** [comment]
air-hun well I think we are going to be out of luck in that case if its a system level update

**Comment by vegerfrancois** [comment]
Hello Marcelino 80,you first have to look in settings and than you must find(find my device) and un thick that.Now you can remove updates from google play services if you thick on the 3 dots.
(sorry for my bad englisch,Iam from Holland.
Find it in deviceadministrator,by other securety settings, I do not know where it is in Google pixel 1,but it is there. Good luck

**Comment by jdotdillinger** [comment]
Thanks vegerfrancois and Marcelino 80. I was unable to uninstall Google play services then reinstall the mentioned prefious version. Going to test it in the car now.

**Comment by yousifabro** [comment]
Still not working on s8 for me. Uninstalled updates on google services and installed 12.2.21 as well as older versions such as 12.2.13. Android auto is on 2.9. Is there anything else that i'm missing?

**Comment by ernstgot** [comment]
Once I uninstalled the updated google services and installed the 12.2.21 my phone keeps saying google play services keeps stopping and other apps keep saying they are not able to run. any ideas?

**Comment by yousifabro** [comment]
Thank you Marcelino, completely forgot to set that again. I appreciate the help.

**Comment by michelesilla** [comment]
After that can I reactivate “find my device “‘in device manager administrators?

Inviato da iPhone

**Comment by michelesilla** [comment]
So... It works!! thk to Zetrox2k

**Comment by michelesilla** [comment]
If I Want upgrade google play service, how I can do? I’m working with 12.2.51 e AA works fine but there is a way to upgrade google play service automatically from smartphone?

Inviato da iPhone

**Comment by jdotdillinger** [comment]
So my Google play services updated itself after I down graded. What can I do?

**Comment by michelesilla** [comment]
Why?

Inviato da iPhone

**Comment by air-hun** [comment]
jdotdillinger:
I hate to say this, but I've said.

**Comment by noizergd** [comment]
I've also tried everything but it seems that AA doesn't want to start without the latest version of Googles Play services. Any other issue? i've an Samsung S6 edge.

**Comment by noizergd** [comment]
update: don't know how but it seems that clicking on update google play services just after installing the version 12.2.21 doesn't bring us to the version that creates the problem.

i've just upadted google play services (after having the notification that AA won't start without updating google play services) and i'm still in the version 12.2.21.

i've juste come back from my auto and it works now!

thanks all for your back up/

**Comment by michelesilla** [comment]
You can try other 2.9 version... there are available some variants for each CPU ( smartphone type). Search apk from google

Inviato da iPhone

**Comment by eurojet69** [comment]
yep that did not work either , tried both ARM and ARM64 (mine is 64) and even went down lower to the previous version  and that did not work either.  I know it is hard troubleshooting via chat 
really loved it , due to my trailer back up camera on my phone screen being able to see it on the larger screen

**Comment by noizergd** [comment]
Update: since rebooting my s6 edge, i m forced to install vetsion 12.5.21 with the problem that everything disappear in AA 2.9

**Comment by noizergd** [comment]
I confirm google play services comes automatic back to version 12.5.21 even when i shut the internet connection down during the night. Dont know how google can do that...

**Comment by noizergd** [comment]
Seems there is no solutions at all. Every previous version of google play services crashes and reboot.. damn

**Comment by i040628** [comment]
I did not find a solution but a workaround:) : using a tablet. Not the same but it has some advantage as well like hopefully bigger screen and being able to use navigation on the car screen and watching video on tablet at the same time.

**Comment by cupral** [comment]
Is it possible, with apk editor, to change the google play services version? If it is, we could create a custom playservice.apk with a very high version, like 20 and probably the phone will not try to update anymore...

**Comment by michelesilla** [comment]
Samsung s8.oreo 8.0
Downgrade to 12.2.21 from 12.5.21.

Autoupdate disabled . NO AUTO UPDATE. All warking !

you just have to read up thread!!! Disable find my phone. Enter play store and disable autoupdate
Return in my find my phone end turn it ON
then Login gmail account from system -google-.

Done!



Inviato da iPhone

**Comment by IanXTsai** [comment]
How to force to don't update 12.2.21 to 12.5.21 again automatically?
Downgrade to 12.2.21  just ok for 1 day... google play services will auto update to 12.5.21.

**Comment by jdotdillinger** [comment]
@michelesilla what happens when you restart your phone?

**Comment by michelesilla** [comment]
Restart system files.
My google play servives is blocked. 
I have 12.2.21 fixed. No one update it from 1 week

Inviato da iPhone

**Comment by noizergd** [comment]
Also go to parameter of your phone-->about-->info logiciel--> tip a lot on version number so that you enter de developer mode.

Once that is done, go back and enter development option and disable system auto update.

Normally will disable auto update of google play services.

Let me know if it is well the case. It works for me.

**Comment by Inkathu** [comment]
Does anybody know the approch which google uses to block the application? Maybe it's possible to do the following
1) Obfuscate or refactor source code
2) creater new build
3) try to use
It could help in case if google doesn't use some kind of API blocking.

**Comment by ajdg29** [comment]
So I have tried everything that has been suggested above and what I've found elsewhere to stop Google play services from updating from 12.2.21, including the above post.  Woke up this morning to find it auto updated to 12.5.21 still. I don't get it. I know there is a way with lucky patcher from a YouTube videobibsaw, but the video is a little older, so the steps aren't exactly on the video and I don't really want to make a big mess of my phone.

**Comment by IanXTsai** [comment]
It may be a temp solution, auto down grade to 12.2.21 everyday?

**Comment by michelesilla** [comment]
I use samsung s8, i’ve downgrade 1 week ago and still 12.2.51. Works . No autouodate!!

Inviato da iPhone

**Comment by ajdg29** [comment]
@michelesilla ,

Do you have a unlocked/rooted phone?  I have a Samsung S8+.  I have been trying to keep the google play services from auto-updating for the past week or so, and have tried a ton of things as noted above.  Do you have 12.2.51 as noted above, or is it 12.2.21?  Just trying to figure what you are doing differently that I haven't tried.

**Comment by michelesilla** [comment]
I’ m using 12.2.21 goole play services
I’ve disable find my phone .
I’ve unistall 12.5.21
I’ve reboot device
I’ve reinstall 12.2.21 for my device armv8

First all that i’ve turned on administrator development and disable system autoupdate

Turned on : usb debug ( for manually install AA from terminal)

I’ve disable autoupdate from play store

At last after 1 day i’ve turned on find my phone.



This is just What I do







Inviato da iPhone

**Comment by michelesilla** [comment]
NO ROOT

Inviato da iPhone

**Comment by DavidHunag** [comment]
I would like to further explain the part of disable autoupdate from play store:
Enter the settings → Notifications → uncheck the "updates" and "Auto-updates".
Done!
So far, the GPS never update or noising pop-up information.
But, if you reboot your phone, GPS will auto update to the last vision.
For you reference.

**Comment by cupral** [comment]
This worked for me (root required):
https://forum.xda-developers.com/android/help/delete-disable-downgrade-prevent-update-t3091832
